iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp

```diff
@@ -0,0 +1,672 @@
+        })
+        .or_else([&]() { GTEST_FAIL() << "Expected STOP_OFFER message but got none"; });
+    EXPECT_FALSE(interfacePort->m_caproMessageFiFo.pop().has_value());
+}
+
+TEST_F(PortManager_test, ServerStateIsForwardedToInterfacePortWhenDestroyed)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3e9660f8-046c-4e3a-acfd-bad33a6f999c");
+    auto interfacePort = m_portManager->acquireInterfacePortData(iox::capro::Interfaces::DDS, "penguin");
+    auto serverOptions = createTestServerOptions();
+    serverOptions.offerOnCreate = true;
+    auto serverPortUser = createServer(serverOptions);
+    m_portManager->doDiscovery();
+
+    // empty fifo
+    while (interfacePort->m_caproMessageFiFo.pop().has_value())
+    {
+    }
+
+    serverPortUser.destroy();
+    m_portManager->doDiscovery();
+
+    interfacePort->m_caproMessageFiFo.pop()
+        .and_then([&](const auto& caproMessage) {
+            EXPECT_THAT(caproMessage.m_type, Eq(CaproMessageType::STOP_OFFER));
+            EXPECT_THAT(caproMessage.m_serviceType, Eq(CaproServiceType::SERVER));
+        })
+        .or_else([&]() { GTEST_FAIL() << "Expected STOP_OFFER message but got none"; });
+    EXPECT_FALSE(interfacePort->m_caproMessageFiFo.pop().has_value());
+}
+
+TEST_F(PortManager_test, ServerStateIsForwardedToInterfacePortWhenAlreadyOfferAndInterfacePortIsNewlyCreated)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "31563bb9-561c-43ee-8e3e-b6676cfc9547");
+    auto serverOptions = createTestServerOptions();
+
+    serverOptions.offerOnCreate = true;
+    auto serverPortUser = createServer(serverOptions);
+
+    m_portManager->doDiscovery();
+
+    auto interfacePort = m_portManager->acquireInterfacePortData(iox::capro::Interfaces::DDS, "penguin");
+    m_portManager->doDiscovery();
+
+    interfacePort->m_caproMessageFiFo.pop()
+        .and_then([&](const auto& caproMessage) {
+            EXPECT_THAT(caproMessage.m_type, Eq(CaproMessageType::OFFER));
+            EXPECT_THAT(caproMessage.m_serviceType, Eq(CaproServiceType::SERVER));
+        })
+        .or_else([&]() { GTEST_FAIL() << "Expected OFFER message but got none"; });
+    EXPECT_FALSE(interfacePort->m_caproMessageFiFo.pop().has_value());
+}
+
+// END forwarding to InterfacePort tests
+
+// BEGIN policy based connection tests
+
+// NOTE: there is a client/server sandwich to test both code paths where the client and
+// the server initiates the state machine ping pong
+
+TEST_F(PortManager_test, ClientWithDiscardOldestDataAndServerWithDiscardOldestDataAreConnected)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "56871f9d-d7c1-4c3c-b86c-9a1e1dc9fd74");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.responseQueueFullPolicy = QueueFullPolicy::DISCARD_OLDEST_DATA;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.clientTooSlowPolicy = ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA;
+
+    auto clientPortUser1 = createClient(clientOptions);
```

Comment:
Why are you using always 2 client ports in those tests. It is fine by me I was just wondering since I thought one client port should suffice.